src/Client/Pages/Identity/Roles/RolePermissions.razor.cs

```diff
@@ -39,42 +38,30 @@ protected override async Task OnInitializedAsync()
         _canEditRoleClaims = (await AuthService.AuthorizeAsync(state.User, FSHPermissions.RoleClaims.Edit)).Succeeded;
         _canSearchRoleClaims = (await AuthService.AuthorizeAsync(state.User, FSHPermissions.RoleClaims.View)).Succeeded;
 
-        var rolePermissions = new List<PermissionUpdateDto>();
-
         if (await ApiHelper.ExecuteCallGuardedAsync(
                 () => RolesClient.GetByIdWithPermissionsAsync(Id), Snackbar)
             is RoleDto role)
         {
-            Title = role.Name;
-            Description = string.Format(_localizer["Manage {0}'s Permissions"], role.Name);
-
-            rolePermissions = role.Permissions?.Adapt<List<PermissionUpdateDto>>();
-        }
-
-        var allPermissions = DefaultPermissions.Admin;
-        allPermissions.AddRange(DefaultPermissions.Root);
-        var result = allPermissions.Select(x => new PermissionUpdateDto()
-        {
-            Permission = x
-        }).ToList();
-
-        foreach (var permission in result)
-        {
-            if (rolePermissions.Select(z => z.Permission).Contains(permission.Permission))
-                permission.Enabled = true;
+            _title = string.Format(_localizer["{0} Permissions"], role.Name);
+            _description = string.Format(_localizer["Manage {0} Role Permissions"], role.Name);
```

Comment:
Put these two outside the foreach cycle maybe

Reply:
what foreach cycle?

Reply:
sorry, i as reading the preview wrong xD 
Nice work ;-)